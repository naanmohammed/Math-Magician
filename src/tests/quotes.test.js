import React from 'react';
import {render, act} from '@testing-library/react';
import Quotes from '../components/quotes';
describe('Quotes', () => {
  it('should render loading message while quotes are being fetched', () => {
    const { container } = render(<Quotes />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render failed message if quotes failed to fetch', async () => {
    jest.spyOn(global, 'fetch').mockImplementation(() =>
      Promise.reject(new Error('Failed to fetch quotes'))
    );
    await act(async () => {
      render(<Quotes />);
    });
  });
});