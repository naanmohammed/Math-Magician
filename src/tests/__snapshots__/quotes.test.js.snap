// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes should render loading message while quotes are being fetched 1`] = `
<div
  class="loading"
>
  Loading quotes ...
</div>
`;
